artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/AbstractSequentialFile.java

```diff
@@ -271,7 +271,7 @@ protected ByteBuffer newBuffer(int size, int limit) {
       public void flushBuffer(final ByteBuf byteBuf, final boolean requestedSync, final List<IOCallback> callbacks) {
          final int bytes = byteBuf.readableBytes();
          if (bytes > 0) {
-            final ByteBuffer buffer = newBuffer(byteBuf.capacity(), bytes);
+            final ByteBuffer buffer = newBuffer(byteBuf.capacity(), bytes, false);
```

Comment:
In case we pool a buffer, and say you wrote 300 bytes, on a place that previously had 4K filled

you will fill it with 300 bytes... the buffer will be flipped, but when the libaio write takes place.. I'm afraid this could not cleanup the rest of the 4K.

I did not run any tests on this.. just from what I remember on the pool mechanism.

just make sure the previously pooled buffer wouldn't make to the journal.

Reply:
Also: did you run the whole test suite on such change?

Reply:
Every borrowed pooled buffer always start at position 0, so the observer, while flushing always overwrite exactly the bytes it has requested despite being zeroed or not

> Also: did you run the whole test suite on such change?

Yet to do it, will do it now :)
NIO and MAPPED already use this optimization from long time (the jira link explain it): ASYNCIO was the only one missing yet

Reply:
About the tests: I see that only this one has failed

```
Error Message
expected:<10> but was:<11>
Stacktrace
java.lang.AssertionError: expected:<10> but was:<11>
	at org.apache.activemq.artemis.tests.unit.core.postoffice.impl.DuplicateDetectionUnitTest.testReloadDuplication(DuplicateDetectionUnitTest.java:127)
```
And probably it isn't related

Reply:
I actually had access to the same build as you, and as I can see the build did not complete.. lots of failures due to open files. I will restart that build

Reply:
@franz1981 the failure is real... this PR is broken.